### PR TITLE
Allow for gmsh mesh format 4.1

### LINKED ISSUE #8245: GMSH-4.3 msh compatibility (Mesh format 4.1)
We should update our `read_msh` function to be able to deal with meshes created by gmsh-4.3:

```
2: --------------------------------------------------------
2: An error occurred in line <1479> of file <../source/grid/grid_in.cc> in function
2:     void dealii::GridIn<dim, spacedim>::read_msh(std::istream&) [with int dim = 2; int spacedim = 2; std::istream = std::basic_istream<char>]
2: The violated condition was: 
2:     n_physicals < 2
2: Additional information: 
2:     More than one tag is not supported!
2: --------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #8245.

### CODE REVIEW

File: doc/news/changes/minor/20190521DanielArndt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Improved: GridIn::read_msh has been extended to alow for gmsh mesh format 4.1.
```
Comment (kronbichler) [inline doc/news/changes/minor/20190521DanielArndt]:
```suggestion
Improved: GridIn::read_msh has been extended to allow for gmsh mesh format 4.1.
```

**Comment by kronbichler** [comment]
@tamiko do we want with the release for this PR? It is a lot of changes, at the same time I assume that @masterleinad has properly tested both paths, so I'd be confident.

Now I see that @luca-heltai has an alternative #8266 - can the two of you @masterleinad and @luca-heltai agree on one version?

**Comment by luca-heltai** [comment]
I was late... :) This looks better.

**Comment by masterleinad** [comment]
> I was late... :) This looks better.

I would also be fine with your version. It seems that we are introducing the same changes.
I didn't test this extensively, though. I was just comparing the two mesh format descriptions and making sure the test passes.

**Comment by luca-heltai** [comment]
I checked with both gmsh 4.2.2 and 4.0.0, and everything looks fine.

**Comment by tamiko** [comment]
@kronbichler I really want this in 9.1

**Comment by tamiko** [comment]
The testsuite passed the build stage and is now running tests (none of which have gmsh enabled).

**Pull request merged**